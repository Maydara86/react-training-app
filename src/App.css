body {
  margin: 0;
  height: 100%;
  padding: 10px;
}

.articleSummaryGridContainer {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  grid-template-rows: 1fr 1fr;
  grid-auto-columns: 1fr 1fr 1fr;
  grid-gap: 1em 1em;
  place-items: center center;
  place-content: center center;
}

.ListContainer {
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-align-items: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 10px;
}

.ListItem:nth-child(1) {
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
}