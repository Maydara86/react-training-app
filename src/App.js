import React from 'react'
import './App.css'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
// import ArticleSummaryContainer from './components/Article/ArticleSummaryContainer'
import ArticleSummaryListItem from './components/ArticlesList/ArticleSummaryListItem'
import ArticleSummaryGridCard from './components/ArticleGrid/ArticleSummaryGridCard'
import UserWithFollowButton from './components/User/UserWithFollowButton'
import clickStarHandler, { clickFollowHandler } from './store/actions/usersAction'
import clickBookmarkHandler, { clickClapHandler } from './store/actions/articlesAction'
import usersSelector from './selectors/users'
import articlesSelector from './selectors/articles'

function App(props) {
  /* eslint-disable no-shadow */
  const {
    users,
    articles,
    clickClapHandler,
    clickBookmarkHandler,
    clickStarHandler,
    clickFollowHandler,
  } = props
  /* eslint-disable no-shadow */

  return (
    <div>
      {/* {users.map(user => {
        return (
          <UserWithFollowButton
            user={user}
            key={user.id}
            handleFollowClick={clickFollowHandler}
            handleStarClick={clickStarHandler}
          />
        )
      })} */}
      {/* <div className="articleSummaryGridContainer">
        {articles.map((article, i) => {
          return (
            <ArticleSummaryGridCard
              key={article.id}
              {...article}
              user={users[i]}
              handleClapClick={clickClapHandler}
              handleBookmarkClick={clickBookmarkHandler}
              handleFollowClick={clickFollowHandler}
              handleStarClick={clickStarHandler}
            />
          )
        })}
      </div> */}

      <div className="ListContainer">
        {articles.map((article, i) => {
          return (
            <ArticleSummaryListItem
              className="ListItem"
              key={article.id}
              {...article}
              user={users[i]}
              handleClapClick={clickClapHandler}
              handleBookmarkClick={clickBookmarkHandler}
              handleFollowClick={clickFollowHandler}
              handleStarClick={clickStarHandler}
            />
          )
        })}
      </div>
    </div>
  )
}

const mapStateToProps = state => {
  return {
    articles: articlesSelector(state),
    users: usersSelector(state),
  }
}

const mapDispatchToProps = dispatch => {
  return bindActionCreators(
    { clickStarHandler, clickBookmarkHandler, clickClapHandler, clickFollowHandler },
    dispatch
  )
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App)
