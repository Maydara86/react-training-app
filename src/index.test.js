import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from './App';
import * as serviceWorker from './serviceWorker';

it('renders the App component on the DOM', () => {
    const tree = renderer.create(<App />).toJSON()
    expect(tree).toMatchSnapshot()
})

test('test the ReactDOM.rendrer function', () => {
    const spy = jest.spyOn(ReactDOM, 'render');
    // const isRendering = ReactDOM.render();
  
    expect(spy).toHaveBeenCalled();
    expect(ReactDOM.render()).toBe(true);
  
    spy.mockRestore();
});

test('test the serviceWorker unregister function', () => {
    const spy = jest.spyOn(serviceWorker, 'unregister');
    // const unregistering = serviceWorker.unregister();
  
    expect(spy).toHaveBeenCalled();
    expect(serviceWorker.unregister()).toBe(true);
  
    spy.mockRestore();
});