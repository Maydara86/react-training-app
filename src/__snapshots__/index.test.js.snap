// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the App component on the DOM 1`] = `
<div>
  <div
    className="container"
  >
    <img
      alt="Walton Morton"
      className="imageContainer"
      src="avatar.png"
    />
    <div
      className="contentContainer"
    >
      <div
        className="contentContainerRow1"
      >
        <div
          className="name"
        >
          name: 
          Walton Morton
        </div>
        <button
          onClick={[Function]}
        >
          Unfollow
        </button>
      </div>
      <div
        className="contentContainerRow2"
      >
        <div
          className="date"
        >
          date: 
          Aug 9
        </div>
        <div
          className="time"
        >
          reading time: 
          20
        </div>
        <input
          checked={false}
          className="hvrIconPop"
          onChange={[Function]}
          type="checkbox"
        />
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <img
      alt="Joy Dennis"
      className="imageContainer"
      src="avatar.png"
    />
    <div
      className="contentContainer"
    >
      <div
        className="contentContainerRow1"
      >
        <div
          className="name"
        >
          name: 
          Joy Dennis
        </div>
        <button
          onClick={[Function]}
        >
          Follow
        </button>
      </div>
      <div
        className="contentContainerRow2"
      >
        <div
          className="date"
        >
          date: 
          Aug 9
        </div>
        <div
          className="time"
        >
          reading time: 
          3
        </div>
        <input
          checked={true}
          className="hvrIconPop"
          onChange={[Function]}
          type="checkbox"
        />
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <img
      alt="Frances Smith"
      className="imageContainer"
      src="avatar.png"
    />
    <div
      className="contentContainer"
    >
      <div
        className="contentContainerRow1"
      >
        <div
          className="name"
        >
          name: 
          Frances Smith
        </div>
        <button
          onClick={[Function]}
        >
          Follow
        </button>
      </div>
      <div
        className="contentContainerRow2"
      >
        <div
          className="date"
        >
          date: 
          Aug 9
        </div>
        <div
          className="time"
        >
          reading time: 
          14
        </div>
        <input
          checked={false}
          className="hvrIconPop"
          onChange={[Function]}
          type="checkbox"
        />
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <img
      alt="Erma Kirby"
      className="imageContainer"
      src="avatar.png"
    />
    <div
      className="contentContainer"
    >
      <div
        className="contentContainerRow1"
      >
        <div
          className="name"
        >
          name: 
          Erma Kirby
        </div>
        <button
          onClick={[Function]}
        >
          Unfollow
        </button>
      </div>
      <div
        className="contentContainerRow2"
      >
        <div
          className="date"
        >
          date: 
          Aug 9
        </div>
        <div
          className="time"
        >
          reading time: 
          9
        </div>
        <input
          checked={true}
          className="hvrIconPop"
          onChange={[Function]}
          type="checkbox"
        />
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <img
      alt="Burgess Webb"
      className="imageContainer"
      src="avatar.png"
    />
    <div
      className="contentContainer"
    >
      <div
        className="contentContainerRow1"
      >
        <div
          className="name"
        >
          name: 
          Burgess Webb
        </div>
        <button
          onClick={[Function]}
        >
          Follow
        </button>
      </div>
      <div
        className="contentContainerRow2"
      >
        <div
          className="date"
        >
          date: 
          Aug 9
        </div>
        <div
          className="time"
        >
          reading time: 
          3
        </div>
        <input
          checked={true}
          className="hvrIconPop"
          onChange={[Function]}
          type="checkbox"
        />
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <img
      alt="Britney Stephenson"
      className="imageContainer"
      src="avatar.png"
    />
    <div
      className="contentContainer"
    >
      <div
        className="contentContainerRow1"
      >
        <div
          className="name"
        >
          name: 
          Britney Stephenson
        </div>
        <button
          onClick={[Function]}
        >
          Follow
        </button>
      </div>
      <div
        className="contentContainerRow2"
      >
        <div
          className="date"
        >
          date: 
          Aug 9
        </div>
        <div
          className="time"
        >
          reading time: 
          15
        </div>
        <input
          checked={false}
          className="hvrIconPop"
          onChange={[Function]}
          type="checkbox"
        />
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <img
      alt="Kristi Fitzgerald"
      className="imageContainer"
      src="avatar.png"
    />
    <div
      className="contentContainer"
    >
      <div
        className="contentContainerRow1"
      >
        <div
          className="name"
        >
          name: 
          Kristi Fitzgerald
        </div>
        <button
          onClick={[Function]}
        >
          Follow
        </button>
      </div>
      <div
        className="contentContainerRow2"
      >
        <div
          className="date"
        >
          date: 
          Aug 9
        </div>
        <div
          className="time"
        >
          reading time: 
          6
        </div>
        <input
          checked={false}
          className="hvrIconPop"
          onChange={[Function]}
          type="checkbox"
        />
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <img
      alt="Ines Dalton"
      className="imageContainer"
      src="avatar.png"
    />
    <div
      className="contentContainer"
    >
      <div
        className="contentContainerRow1"
      >
        <div
          className="name"
        >
          name: 
          Ines Dalton
        </div>
        <button
          onClick={[Function]}
        >
          Follow
        </button>
      </div>
      <div
        className="contentContainerRow2"
      >
        <div
          className="date"
        >
          date: 
          Aug 9
        </div>
        <div
          className="time"
        >
          reading time: 
          7
        </div>
        <input
          checked={true}
          className="hvrIconPop"
          onChange={[Function]}
          type="checkbox"
        />
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <img
      alt="Terra Hoffman"
      className="imageContainer"
      src="avatar.png"
    />
    <div
      className="contentContainer"
    >
      <div
        className="contentContainerRow1"
      >
        <div
          className="name"
        >
          name: 
          Terra Hoffman
        </div>
        <button
          onClick={[Function]}
        >
          Unfollow
        </button>
      </div>
      <div
        className="contentContainerRow2"
      >
        <div
          className="date"
        >
          date: 
          Aug 9
        </div>
        <div
          className="time"
        >
          reading time: 
          9
        </div>
        <input
          checked={false}
          className="hvrIconPop"
          onChange={[Function]}
          type="checkbox"
        />
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <img
      alt="Mccall Yang"
      className="imageContainer"
      src="avatar.png"
    />
    <div
      className="contentContainer"
    >
      <div
        className="contentContainerRow1"
      >
        <div
          className="name"
        >
          name: 
          Mccall Yang
        </div>
        <button
          onClick={[Function]}
        >
          Unfollow
        </button>
      </div>
      <div
        className="contentContainerRow2"
      >
        <div
          className="date"
        >
          date: 
          Aug 9
        </div>
        <div
          className="time"
        >
          reading time: 
          8
        </div>
        <input
          checked={false}
          className="hvrIconPop"
          onChange={[Function]}
          type="checkbox"
        />
      </div>
    </div>
  </div>
</div>
`;
